internal/lightcone/preservation/wearewildfire/wearewildfire.go

```diff
@@ -0,0 +1,58 @@
+package wearewildfire
+
+import (
+	"github.com/simimpact/srsim/pkg/engine"
+	"github.com/simimpact/srsim/pkg/engine/equip/lightcone"
+	"github.com/simimpact/srsim/pkg/engine/event"
+	"github.com/simimpact/srsim/pkg/engine/info"
+	"github.com/simimpact/srsim/pkg/engine/prop"
+	"github.com/simimpact/srsim/pkg/key"
+	"github.com/simimpact/srsim/pkg/model"
+)
+
+const (
+	mod     key.Modifier = "wearewildfire"
+	modHeal key.Modifier = "wearewildfire-heal"
+)
+
+// At the start of the battle, the DMG dealt to all allies decreases by
+// 8%/10%/12%/14%/16% for 5 turn(s). At the same time, immediately restores HP
+// to all allies equal to 30%/35%/40%/45%/50% of the respective HP difference
+// between the characters' Max HP and current HP.
+func init() {
+	lightcone.Register(key.WeAreWildfire, lightcone.Config{
+		CreatePassive: Create,
+		Rarity:        4,
+		Path:          model.Path_PRESERVATION,
+		Promotions:    promotions,
+	})
+}
+
+func Create(engine engine.Engine, owner key.TargetID, lc info.LightCone) {
+	// team DMG RES
+	amt := 0.06 + 0.02*float64(lc.Imposition)
+	amtHeal := 0.25 + 0.05*float64(lc.Imposition)
+
+	dmgmod := info.Modifier{
+		Name:     mod,
+		Source:   owner,
+		Stats:    info.PropMap{prop.AllDamageReduce: amt},
+		Duration: 5,
+	}
```

Comment:
This mod needs to be registered with the following settings:
* Stacking: `ReplaceBySource` (can have multiple if multiple allies have this LC)
* TickMoment: `ModifierPhase1End`
* StatusType: `BUFF` (this can be found by searching the modifier name in the `ExcelOutput/StatusConfig.json` file in the datamine)

Reply:
what do `ModifierPhase1End` and `ModifierPhase2End` denote exactly ?

Reply:
This field is for when modifiers duration ticks down. Atm we only support `ModifierPhase1End` & `ModifierPhase2End` (where if left unspecified modifiers will default to the 2nd). `Phase1` & `Phase2` refer to a specific time in the turn loop which is when stuff happens. `Phase1` is typically where stuff like DoTs deal damage whereas `Phase2` is at the end of the turn after there are no more actions remaining